include/lsst/afw/geom/SpanSet.h

```diff
@@ -0,0 +1,563 @@
+     * a generic iterator can have any behavior, bouds checking is not possible. As such the iterator
+     * must be valid for at least the number of pixels contained in the SpanSet. Numeric values are
+     * also different in that they will not be iterated but the value will be passed to the functor
+     * for each point in the SpanSet.
+
+     * Because a ndarray can either be interpreted as either a two dimensional image, or a one
+     * dimensional vector a user must break the ambiguity by using the functions ndarray::ndImage,
+     * or ndarray::ndFlat respectively.
+     *
+     * The functor object must operate in-place on any data, no return values are captured. No exceptions
+     * are handled, if a functor throws one it will propagate back out of the applyFunctor call, possibly
+     * leaving the output array in an incomplete state. The first argument of the functor must be
+     * a Point2I which will be the point in the SpanSet where the operation is occurring. All the
+     * remaining arguments will be individual values generated from the input arguments. For two
+     * dimensional types (Image, MaskedImage, ndarray) arguments will be the value at the location given
+     * by the SpanSet, and thus all SpanSet coordinates must be addressable in these types. For one
+     * dimensional types (ndarray, iterator) arguments will correspond to the value at the location defined
+     * by number of pixels which have been iterated over. The total length of a one dimensional type
+     * will be equal to the area of the SpanSet, and therefore the data-type must be at least that length.
+     * The order of the parameters supplied to the functor will be the same order as they are passed to
+     * the applyFunctor method.
+     *
+     * @param ...args - Variadic arguments, may be of type Image, MaskedImage, ndarrays, numeric values
+     *                  and iterators. ndarrays must be specified as either image or vector with the
+     *                  ndarray::ndImage or ndarray::ndFlat
+     *
+     * @tparam ...Args - Variadic type specification
+     */
+    template <typename Functor, typename...Args>
+    void applyFunctor(Functor && func, Args && ...args) const {
+        /* Use a variadic template to take a functor object, and an arbitrary number
+           of parameters. For each of the arguments, construct a Getter class using
+           a function (makeGetter) which is overloaded to all the types applyFunctorImpl
+           supports: Images, MaskedImages, Exposures, ndarrays, numeric values, and
+           iterators. The functor and the getters are then passed to the implementation of
+           applyFunctor where the values of the input arguments are intelligently
+           generated at each point in SpanSet, and passed to the functor object for evaluation.
+         */
+        applyFunctorImpl(func, details::makeGetter(args)...);
```

Comment:
No, that will be changed. Jim and I tired a few different things but could not get the syntax of forwarding to work quite right with the variadic arguments. I chose to not keep beating my head against it and just wait until Pim got back to work it out the forwarding. It is already fixed in my working copy.